lib/cisco_node_utils/cisco_cmn_utils.rb

```diff
@@ -107,18 +107,30 @@ def self.process_network_mask(network)
       network
     end
 
+    def self.depth(a)
+      return 0 unless a.is_a?(Array)
+      1 + depth(a[0])
+    end
+
     # Helper to build a hash of add/remove commands for a nested array.
     # Useful for network, redistribute, etc.
     #   should: an array of expected cmds (manifest/recipe)
     #  current: an array of existing cmds on the device
     def self.delta_add_remove(should, current=[])
       # Remove nil entries from array
-      should.each(&:compact!) unless should.empty?
+      should.each(&:compact!) unless should.empty? if depth(should) > 1
```

Comment:
Please add more comments - it's not at all obvious to me why we're checking the 'depth' of an array.